docs/examples/01_current_injection.py

```diff
@@ -1,502 +1,694 @@
-# fig
-# # %%
-# # Now:
-# #
-# # - compare number of spikes within each current interval
-# #
-# # - plot each current interval, show how rate is linear with input and so misses some effects
-# #
-# # - show tuning curves, point out that they differ at high input levels
-
-
-# # point out this is two parameters and is only using input at time t, say we
-# # can get a little fancier by adding history. try with and without bases?
-# #
-# # then add self-excitatin
+# %%
+#
+# In addition to making that mismatch discussed earlier a little more obvious,
+# this tuning curve comparison also highlights that this model thinks the
+# firing rate will continue to grow as the injected current increases, which is
+# not reflected in the data.
+#
+# Viewing this plot also makes it clear that the model's tuning curve is an
+# exponential. We already knew that! That's what it means to be a LNP model of
+# a single input. But it's nice to see it made explicit.
+#
+# ### Finishing up
+#
+# There are a handful of other operations you might like to do with the GLM.
+# First, you might be wondering how to simulate spikes -- the GLM is a LNP
+# model, but the firing rate is just the output of *LN*, its first two steps.
+# The firing rate is just the mean of a Poisson process, so we can pass it to
+# `jax.random.poisson`:
+
+spikes = jax.random.poisson(jax.random.PRNGKey(0), predicted_fr.values)
+
+# %%
+#
+# Note that this is not actually that informative and, in general, it is
+# recommended that you focus on firing rates when interpreting your model. In
+# particular, if your GLM includes auto-regressive inputs (e.g., neurons are
+# connected to themselves or each other), simulate can behave poorly, see
+# **XXX** for details.
+#
+# Secondly, you may want a number with which to evaluate your model's
+# performance. As discussed earlier, the model optimizes log-likelihood to find
+# the best-fitting weights, and we can calculate this number using its `score`
+# method:
+
+log_likelihood = model.score(predictor, count, score_type="log-likelihood")
+print(f"log-likelihood: {log_likelihood}")
+
+# %%
+#
+# This log-likelihood is un-normalized and thus doesn't mean that much by
+# itself, other than "higher=better". When comparing alternative GLMs fit on
+# the same dataset, whether that's models using different regularizers and
+# solvers or those using different predictors, comparing log-likelihoods is a
+# reasonable thing to do.
+#
+# !!! info
+#
+#     Under the hood, nemos is minimizing the negative log-likelihood, as is
+#     typical in many optimization contexts. `score` returns the real
+#     log-likelihood, however, and thus higher is better.
+#
+# Because it's un-normalized, however, the log-likelihood should not be
+# compared across datasets (because e.g., it won't account for difference in
+# noise levels). We provide the ability to compute the pseudo-$R^2$ for this
+# purpose:
+
+model.score(predictor, count, score_type='pseudo-r2-Cohen')
```

Comment:
I think this part might want to be placed into its own section called "Evaluating the model on train and test sets"

Reply:
Is this section still part of the main text or placed under "Further exercises" with the "Train and test sets" bit?